scripts/update_rif_table.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (C) University of Tennessee Health Science Center, Memphis, TN.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# This program is available from Source Forge: at GeneNetwork Project
+# (sourceforge.net/projects/genenetwork/).
+#
+# Contact Drs. Robert W. Williams and Xiaodong Zhou (2010)
+# at rwilliams@uthsc.edu and xzhou15@uthsc.edu
+#
+#
+#
+# This module is used by GeneNetwork project (www.genenetwork.org)
+#
+# Created by GeneNetwork Core Team 2010/08/10
+# Updated on Lei Yan 2011/02/08
+# created by Lei Yan 02/08/2011
+
+"""
+Script responsible for updating the GenerRIF_BASIC table
+"""
+
+import argparse
+import csv
+import gzip
+import pathlib
+from tempfile import TemporaryDirectory
+from typing import Dict, Generator
+import requests
+
+from gn3.db_utils import database_connection
+
+TAX_IDS = {"10090": 1, "9606": 4, "10116": 2, "3702": 3}
+
+GENE_INFO_URL = "https://ftp.ncbi.nlm.nih.gov/gene/DATA/gene_info.gz"
+GENERIFS_BASIC_URL = "https://ftp.ncbi.nih.gov/gene/GeneRIF/generifs_basic.gz"
```

Comment:
*thought:* avoid creating global variables if they are used once.  The point of global variables is to share some (preferably immutable) state across different functions/modules.  In this case,  I see we only ever use these URLs inside one part (when downloading data), so I recommend you have just directly pass the URL to the function; perhaps with a comment should you feel the need to add one.

Reply:
I view these as more like constants, here's the PEP for constants: https://peps.python.org/pep-0008/#constants

It also provides an easy place to change the endpoints in future.